utilities/backup/backup_engine.cc

```diff
@@ -1432,7 +1433,7 @@ IOStatus BackupEngineImpl::CreateNewBackupWithMetadata(
               live_dst_paths, backup_items_to_finish, new_backup_id,
               false /* shared */, "" /* src_dir */, fname,
               EnvOptions() /* src_env_options */, rate_limiter, type,
-              contents.size(), db_options.statistics.get(), 0 /* size_limit */,
+              contents.size(), db_options.statistics.get(), contents.size(),
```

Comment:
I doubt whether this change is needed for this PR for the following reasons:
(1) backup_engine_test passes with reverting this change 
(2) Based on the semantics, `ReadFileAndComputeChecksum` seems to fall under `AddBackupFileWorkItem` of `copy_file_cb` path (https://github.com/facebook/rocksdb/pull/9973/files#diff-4d85a375a1fe9d14d0e5cfa18bf19729e614a4f3edf73ca59b0e7bde0b1e79c6R1428) instead of `create_file_cb` path (this).  So the change here does not seem to affect the `ReadFileAndComputeChecksum` 
(3) Instead, the [one](https://github.com/facebook/rocksdb/blob/d110e17ffb2c1f25990108de814717f99a329a8b/utilities/backup/backup_engine.cc#L1978) under `VerifyBackup` might be more concerning in terms of over-charging, although the current test does not reveal that. It's probably due to currently the test that accidentally revealed overcharging was written (by me) for cases involving both read and write rate-limiting while `VerifyBackup` is a pure read read-limiting.

But tracing through the test `TEST_P(BackupEngineRateLimitingTestWithParam, RateLimitingVerifyBackup)` + sum up the charged bytes `buffer_to_read` (https://github.com/facebook/rocksdb/blob/d110e17ffb2c1f25990108de814717f99a329a8b/utilities/backup/backup_engine.cc#L2374) should reveal that. 

If we really want to make sure your improvement to prevent overcharging is really helping, then maybe we should add some unit tests to compare "the exact charged request bytes (i.e, RateLimiter::GetTotalBytesThrough())" VS "the estimated actual read bytes" (how to get this info in test is a bit tricky - maybe Env::GetFileSize()?). I expect to see a relatively big gap between this two before your improvement and a smaller gap (if not exactly equal) after your improvement.

For (3), let me double check @cbi42  - is `TEST_P(BackupEngineRateLimitingTestWithParam, RateLimitingChargeReadInRestore)` the only test that failed without your improvement, which I believe can be solved by changing the test to compare `GetTotalRequests` instead of `GetTotalBytesThrough`? If that's the case, I wouldn't mind saving your over-charging improvement into a new PR with unit-test and gets the tests passing first for this PR (considering it's already big). @ajkr  - do you have an opinion on my suggestion?

Btw, @pdillinger can confirm on my understanding about BackupEngine in (2)/(3) - some quick context for you Peter: @cbi42 is working on making `BackupEngine` cap `SequentialFileReader::Read()` by file size so that the rate-limiting made inside SequentialFileReader by this PR does not charge more than file size (i.e, the over-charging problem). And `SequentialFileReader::Read()`  is called in `BackupEngineImpl::ReadFileAndComputeChecksum` and `BackupEngineImpl::CopyOrCreateFile`

Reply:
Btw, @pdillinger can confirm on my understanding about BackupEngine in (2)/(3) - some quick context for you Peter: @cbi42 is working on making BackupEngine cap SequentialFileReader::Read() by file size so that the rate-limiting made inside SequentialFileReader by this PR does not charge more than file size (i.e, the over-charging problem). And SequentialFileReader::Read() is called in BackupEngineImpl::ReadFileAndComputeChecksum and BackupEngineImpl::CopyOrCreateFile